### DISCUSSION TITLE
TypeORM is legacy now - Why?

### DISCUSSION THREAD

**Discussion by Jernik** [nextauthjs/next-auth#4287] [Help] [answered]
### What is the improvement or update you wish to see?

The `next-auth` docs are unclear as to why the TypeORM adapter is legacy. They currently state that "In the future, we might split up this adapter to support single flavors of SQL for easier maintenance and reduced bundle size.", but that doesn't seem like a reason to mark it as a legacy adapter. Should I not use TypeORM and so not use any flavor of sql? The docs are very confusing as to what I should use when I want to use SQL

### Is there any context that might help us understand?

I want to use postgres or mysql as I already have those running on my host, but I am unsure as to what adapter I should use, or if I should use a different database.

### Does the docs page already exist? Please link to it.

https://next-auth.js.org/adapters/typeorm

**Comment by ThangHuuVu** [comment, marked as answer]
You can read the announcement of the TypeORM team here for more context: https://github.com/typeorm/typeorm/issues/3267

**Reply by Jernik** [reply]
While that provides some context for why TypeORM is marked as legacy, it doesn't explain what adapter I should use instead of TypeORM for sql databases. Is there another adapter I should use, or just stick with the legacy adapter?

**Reply by ThangHuuVu** [reply]
The TypeORM adapter AFAIK still works fines. Personally, I'd recommend the Prisma adapter for the DX and the community.

Disclaimer: I'm a Prisma Ambassador.

**Reply by Jernik** [reply]
Thanks! I'll check that out! I still think the TypeORM docs need more context, because the warning it states is a very confusing one.
